.github/workflows/ci.yml

```diff
@@ -0,0 +1,106 @@
+name: CI
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - prod-*
+      - stable-*
+
+  pull_request:
+    branches:
+      - main
+      - master
+      - prod-beta
+      - prod-stable
+      - stage-beta
+      - stage-stable
+env:
+  BRANCH: ${{ github.base_ref }}
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 18
+
+      - name: Cache node modules
+        id: cache-npm
+        uses: actions/cache@v4
+        env:
+          cache-name: cache-node-modules
+        with:
+          # npm cache files are stored in `~/.npm` on Linux/macOS
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Lint
+        run: npm run lint
+
+  units:
+    name: Units - ${{ matrix.node-version }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version:
+          - '18'
```

Comment:
Don't want to hard code this version, see ${{ matrix.node-version }}

Reply:
This is the definition of `matrix.node-version`. It looks funny because it is reference in thae `name` field, but that's how `matrix` works.